baremetal/metal3machine_manager.go

```diff
@@ -1258,6 +1269,8 @@ func (m *MachineManager) SetNodeProviderID(ctx context.Context, bmhID, providerI
 func (m *MachineManager) SetProviderID(providerID string) {
 	m.Metal3Machine.Spec.ProviderID = &providerID
 	m.Metal3Machine.Status.Ready = true
+	m.SetConditionMetal3MachineToTrue(capm3.Metal3MachineNodeReadyCondition)
```

Comment:
just curious, why not to directly use `conditions.MarkTrue(m.Metal3Machine, t)` ?

Reply:
In this particular case I could do what you're recommending, but in other places the interface -- `MachineManagerInterface` --  is utilized instead of the underlying type -- `MachineManager`.